linera-service/src/node_service.rs

```diff
@@ -343,6 +345,41 @@ where
     }
 }
 
+// What follows is a hack to add a chain_id field to `ChainStateView` based on
+// https://async-graphql.github.io/async-graphql/en/merging_objects.html
+
+struct ChainStateViewExtension<C>(Arc<ChainStateView<C>>);
```

Comment:
As long as it's only about the chain ID, we could simplify this:

```suggestion
struct ChainStateViewExtension(ChainId);
```

But I guess the full view will be useful if we extend this further.

Reply:
That's a very good point thanks! Even `C` contains the chain id in the case :)